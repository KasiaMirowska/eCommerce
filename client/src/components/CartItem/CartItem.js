import React from 'react';
import './CartItem.styles.scss';


const CartItem = (props) => {
    const {imageUrl, price, name, quantity} = props;
    
    return (
        <div className='cart-item'>
            <img src={imageUrl} alt='item'/>
            <div className='item-details'>
                <span>{name}</span>
                <span>{quantity} x ${price}</span>
            </div>
        </div>
    )
  
}

export default React.memo(CartItem);