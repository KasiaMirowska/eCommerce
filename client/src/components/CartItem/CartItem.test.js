import React from 'react';
import {shallow} from 'enzyme';
import CartItem from '../../components/CartItem/CartItem';


describe('CartItem', () => {
    const props = {
        imageUrl: 'url', 
        price: 3, 
        name: 'someName', 
        quantity: 1
    };

    it('renders as expected', () => {
        expect(shallow(<CartItem {...props}/>)).toMatchSnapshot();
    })
})