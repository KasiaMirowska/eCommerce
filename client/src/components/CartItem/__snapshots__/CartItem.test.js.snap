// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem renders as expected 1`] = `ShallowWrapper {}`;
